Assets/Scripts/Control&Input/InputManager.cs

```diff
@@ -40,54 +56,39 @@ void OnDestroy()
-    /// </summary>
-    /// <param name="inputAction">The name of an inputAction you want the delegate to be invoked by</param>
-    /// <param name="performed">Is the InputAction you want to add delegate to invoked when action is performed or canceled?</param>
-    /// <param name="inputEvent">The inputEvent delegate you want to be invoked</param>
-    /// <returns>Function to remove added listener, typically called within OnDestroy</returns>
-    protected Action FixListeners(string inputAction, bool performed, InputEvent inputEvent)
-    {
-        Action<InputAction.CallbackContext> function = ctx => inputEvent?.Invoke(ctx);
-        if (performed)
-        {
-            playerInput.actions[inputAction].performed += function;
-            return () => playerInput.actions[inputAction].performed -= function;
-        }
-        else
-        {
-            playerInput.actions[inputAction].canceled += function;
-            return () => playerInput.actions[inputAction].canceled -= function;
-        }
-    }
-
     public void AddListeners()
     {
         // Update moveInput
         onMovePerformed += MoveInputPerformed;
         onMoveCanceled += MoveInputCanceled;
-
-        cleanupCalls.Add(FixListeners("Select", true, onSelect));
-        cleanupCalls.Add(FixListeners("Cancel", true, onCancel));
-        cleanupCalls.Add(FixListeners("Move", true, onMovePerformed));
-        cleanupCalls.Add(FixListeners("Move", false, onMoveCanceled));
-        cleanupCalls.Add(FixListeners("LeftTab", true, onLeftTabPerformed));
-        cleanupCalls.Add(FixListeners("RightTab", true, onRightTabPerformed));
+        // Update lookInput
+        onLookPerformed += LookInputPerformed;
+        onLookCanceled += LookInputCanceled;
+        // Subscribe delegates to inputs
+        playerInput.actions["Select"].performed += Select;
+        playerInput.actions["Cancel"].performed += Cancel;
+        playerInput.actions["Move"].performed += Move;
+        playerInput.actions["Move"].canceled += Move;
+        playerInput.actions["LeftTab"].performed += LeftTab;
+        playerInput.actions["RightTab"].performed += RightTab;
+        playerInput.actions["Fire"].performed += Fire;
+        playerInput.actions["Fire"].canceled += Fire;
+        playerInput.actions["Look"].performed += Look;
+        playerInput.actions["Look"].canceled += Look;
+
+        // Imprison mouse
+        if (playerInput.currentControlScheme == "MouseAndKeyboard")
+        {
+            isMouseAndKeyboard = true;
+            Cursor.lockState = CursorLockMode.Locked;
+        }
 
         AddExtraListeners();
     }
 
     public void RemoveListeners()
     {
-        cleanupCalls.ForEach(callback => callback());
-        // Update moveInput
-        onMovePerformed -= MoveInputPerformed;
-        onMoveCanceled -= MoveInputCanceled;
-
+        RemoveAllListeners();
         RemoveExtraListeners();
```

Comment:
While I don't forsee us extending the inputManager class in lieu of the changes, it could be that we want to in the future for some reason, so I'll leave them in as they currently are.

But yes, they are unused at the moment and could in theory be removed if we want to :)